bootstrap/bpdoc/bpdoc_test.go

```diff
@@ -44,3 +44,34 @@ func TestNestedPropertyStructs(t *testing.T) {
 		}
 	}
 }
+
+func TestAllPackages(t *testing.T) {
+	packages, err := AllPackages(pkgFiles, moduleTypeNameFactories, moduleTypeNamePropertyStructs)
+	if err != nil {
+		t.Fatalf("expected no errors, got %s", err)
+	}
+
+	if numPackages := len(packages); numPackages != 1 {
+		t.Fatalf("Expected 1 package, got %q", numPackages)
+	}
+
+	p := packages[0]
+
+	for _, m := range p.ModuleTypes {
+		for _, p := range m.PropertyStructs {
+			noMutatedProperties(t, p.Properties)
```

Comment:
This looks like an assert function. In go readability, assert functions is not recommended on test practices (see go/gocomments#assert). I would recommend  the function notMutatedProperties to return back an error and let the caller decide how to handle the error.

Reply:
Done.